internal/resource-request-operator/offerUpdater.go

```diff
@@ -0,0 +1,114 @@
+package resourcerequestoperator
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	sharingv1alpha1 "github.com/liqotech/liqo/apis/sharing/v1alpha1"
+	"github.com/liqotech/liqo/internal/resource-request-operator/interfaces"
+	"github.com/liqotech/liqo/pkg/discovery"
+)
+
+const requeueTimeout = 5 * time.Second
+
+// OfferUpdater is a component which wrap all ResourceOffer update logic.
+type OfferUpdater struct {
+	queue workqueue.RateLimitingInterface
+	client.Client
+	broadcasterInt interfaces.BroadcasterInterface
+}
+
+// Setup initialize all parameters of the OfferUpdater component.
+func (u *OfferUpdater) Setup(broadcaster interfaces.BroadcasterInterface, k8Client client.Client) {
+	u.broadcasterInt = broadcaster
+	u.Client = k8Client
+	u.queue = workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "Offer update queue")
+}
+
+// Start run the OfferUpdate worker.
+func (u *OfferUpdater) Start(ctx context.Context, group *sync.WaitGroup) {
+	defer u.queue.ShutDown()
+	group.Add(1)
+	defer group.Done()
+	go u.startRunner(ctx, group)
+	<-ctx.Done()
+}
+
+func (u *OfferUpdater) startRunner(ctx context.Context, group *sync.WaitGroup) {
+	group.Add(1)
+	defer group.Done()
+	wait.Until(u.run, 2*time.Second, ctx.Done())
+}
+
+func (u *OfferUpdater) run() {
+	for u.processNextItem() {
+	}
+}
+
+func (u *OfferUpdater) processNextItem() bool {
+	obj, shutdown := u.queue.Get()
+	if shutdown {
+		return false
+	}
+	err := func(obj interface{}) error {
+		defer u.queue.Done(obj)
+		var clusterID string
+		var ok bool
+		if clusterID, ok = obj.(string); !ok {
+			u.queue.Forget(obj)
+			return fmt.Errorf("error getting object %v from OfferUpater queue. It is not a string", obj)
+		}
+		if err := u.updateOffer(clusterID); err != nil {
+			u.queue.AddRateLimited(clusterID)
+			return fmt.Errorf("error during updating ResourceOffer for cluster %s: %w", clusterID, err)
+		}
+		return nil
+	}(obj)
+	if err != nil {
+		klog.Errorf("Error occurred during ResourceOffer update %s", err)
+		return true
+	}
+	klog.Infof("Update cluster %s processed", obj.(string))
+	// requeue after timeout seconds
+	u.queue.AddAfter(obj, requeueTimeout)
+	return true
+}
+
+func (u *OfferUpdater) updateOffer(clusterID string) error {
+	offerList := &sharingv1alpha1.ResourceOfferList{}
+	if err := u.Client.List(context.Background(), offerList, client.MatchingLabels{
+		discovery.ClusterIDLabel: clusterID,
+	}); err != nil {
+		return err
+	}
+	if len(offerList.Items) > 1 {
+		return fmt.Errorf("too much ResourceOffer for cluster %s", clusterID)
+	} else if len(offerList.Items) == 0 {
+		return fmt.Errorf("no ResourceOffer for cluster %s", clusterID)
+	}
```

Comment:
Shouldn't this condition occur the first time, when no resource offers are present yet?